web/src/compose_banner.ts

```diff
@@ -51,8 +51,24 @@ export const CLASSNAMES = {
     user_not_subscribed: "user_not_subscribed",
 };
 
-export function append_compose_banner_to_banner_list(new_row: HTMLElement): void {
-    scroll_util.get_content_element($("#compose_banners")).append(new_row);
+// This function provides a convenient way to add new elements
+// to a banner container. The function accepts a container element
+// as a parameter, to which a banner can be appended.
+// If no container is specified, the function will add the banner
+// to either the edit or compose container.
+export function append_compose_banner_to_banner_list(
+    banner: HTMLElement,
+    $list_container?: JQuery,
+): void {
+    if (!$list_container) {
+        // This code verifies whether an edit form banners container
+        // exists in the DOM. If it does, then we are in edit mode
+        // and should append the banner container to the edit form.
+        $list_container = $("#edit_form_banners").length
+            ? $("#edit_form_banners")
+            : $("#compose_banners");
+    }
```

Comment:
I don't think we want this conditional -- we should just arrange that the caller always passes the `$list_container` element instead -- it's way easier to write correct code that handles all cases that way.

Reply:
Sure, I got it!